client/src/reducers/error.js

```diff
@@ -1,16 +1,15 @@
-import { ERROR_DISMISS } from '../../../common/actionTypes/error';
+import { AUTH_ERROR, ERROR_DISMISS } from '../../../common/actionTypes/error';
 
 const errorReducer = (state = null, action) => {
-  const { type, error, data } = action;
-  if (type === ERROR_DISMISS) {
-    return null;
-  } else if (error) {
-    return error;
-  } else if (data && data.error) {
-    return data.error;
-  }
+  const { type, data } = action;
 
-  return state;
+  switch (type) {
+    case ERROR_DISMISS: return null;
+    case AUTH_ERROR: {
+      return data.error;
+    }
```

Comment:
I think we used this to catch generic errors from different actions. Now it just catches AUTH_ERROR, right?

Reply:
Correct, but we should probably add a scheme for all errors to follow, instead of multiple schemes. As it is now, we can add other errors by matching other cases in the pattern matching. Previously it matched random errors sent from things that were not ours as well, so our own errors were overwritten or spammed away.

Also, we've never had support for multiple errors..

Reply:
Ah, yeah didn't think about that. 
Adding support for several errors should be relatively simple.